File: DIP/DIP_003/C++/DIP_003_creating_a_transparent_watermark.cpp
#include <opencv2/opencv.hpp>
  
using namespace cv;
using namespace std;
 
int main() {
 
    // file name
    char file_name[100];
 
    // read the image and logo
    cv::Mat image=  cv::imread("sophie_turner.jpg", 1); 
    cv::Mat logo=  cv::imread("datahacker.png", 1); 
 
    // our starting coordinates
    int x = image.cols-logo.cols;
    int y = image.rows-logo.rows;
 
    // define image ROI at image bottom-right
    Mat imageROI= image(cv::Rect(x,y, logo.cols, logo.rows));
 
    // here we inverted the color (so all black are now white and all white are now black)
    cv::Mat invSrc =  cv::Scalar::all(255) - logo;
 
    // use the logo as a mask (must be gray-level)
    cv::Mat mask(invSrc);
 
    // insert by copying only at locations of non-zero mask
    invSrc.copyTo(imageROI, mask);
 
    cv::imshow("Image", image); // show the image
    cv::namedWindow("Image"); 
    cv::waitKey(0); // wait for a key pressed
    cv::destroyAllWindows();
 
    // save image
    sprintf(file_name, "added_transperant_logo.jpg");
    imwrite(file_name, image);
 
    return 0;
}

File: DIP/DIP_003/Python/DIP_003_creating_a_transparent_watermark.py
import cv2
import numpy as np

# read the image and logo
image = cv2.imread('003.jpg', cv2.IMREAD_UNCHANGED)
logo  = cv2.imread('datahacker.png', cv2.IMREAD_UNCHANGED)

# here we inverted the color (so all black are now white and all white are now black)
logo = cv2.bitwise_not(logo) 

# extracted the width and height of the image
(hl, wl) = logo.shape[:2] 
(hi, wi) = image.shape[:2]

image = np.dstack([image, np.ones((hi, wi), dtype="uint8") * 255])

overlay = np.zeros((hi, wi, 4), dtype="uint8")
overlay[hi - hl: hi, wi - wl:wi] = logo

# Performing transistion from one color onto another
cv2.addWeighted(overlay, 1, image, 1.0, 0, image)

# save the image
cv2.imwrite("added_transperant_logo.jpg", image)
cv2.imshow("added_transperant_logo", image)
cv2.waitKey()
cv2.destroyAllWindows()